queries.sql

```diff
@@ -8,3 +8,26 @@
  SELECT *  FROM animals  WHERE neutered = true;
  SELECT *  FROM animals  WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg >= 10.4 and weight_kg <= 17.3;
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = '';
+COMMIT;
+BEGIN TRANSACTION;
+TRUNCATE animals;
+ROLLBACK;
```

Comment:
- [ ] Please note that `TRUNCATE` is a DDL statement, so it performs an [implicit commit](https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html) and that means the data of a table can not be rollback after the command `TRUNCATE TABLE`. Therefore I suggest you use `DELETE` to delete all the records from the `animals` table.
**HIT:** Your code could look like this :point_down: 
```sql
   BEGIN;

   DELETE FROM animals;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```